airbyte-ci/connectors/pipelines/pipelines/builds/python_connectors.py

```diff
@@ -76,29 +89,58 @@ async def _provision_builder_container(self, base_container: Container) -> Conta
 
         return builder.with_exec(["pip", "install", "--prefix=/install", "."])
 
+    def _get_build_customization_module(self) -> Optional[ModuleType]:
+        """Import the build_customization.py file from the connector directory if it exists.
+
+        Returns:
+            Optional[ModuleType]: The build_customization.py module if it exists, None otherwise.
+        """
+        build_customization_spec_path = self.context.connector.code_directory / "build_customization.py"
+        if not build_customization_spec_path.exists():
+            return None
+
+        build_customization_spec = importlib.util.spec_from_file_location(
+            f"{self.context.connector.code_directory.name}_build_customization", build_customization_spec_path
+        )
+        build_customization_module = importlib.util.module_from_spec(build_customization_spec)
+        build_customization_spec.loader.exec_module(build_customization_module)
+        self.logger.info("This connector has a build_customization.py file. Using it to customize the build.")
+        return build_customization_module
+
     async def _build_from_base_image(self) -> Container:
         """Build the connector container using the base image defined in the metadata, in the connectorBuildOptions.baseImage field.
 
         Returns:
             Container: The connector container built from the base image.
         """
         base = self._get_base_container()
+        build_customization_module = self._get_build_customization_module()
```

Comment:
Im thinking hooks related functions like `_get_build_customization_module`, and calls to `pre_connector_install` should move to a folder/file named something like `build_hooks` or `build_lifecycle`

e.g.
```py
# pipelines/builds/hooks/_init__.py

def get_build_customization_module(connector: Connector) -> Optional[ModuleType]:
    # ...

def pre_install_hook(connector: Connector, container: Container) -> Container
    build_customization_module = get_build_customization_module(connector)
    if hasattr(build_customization_module, "pre_connector_install"):
            self.logger.info("Adding the pre_connector_install hook to the base")
            container = await build_customization_module.pre_connector_install(container)

    return container
    
def post_install_hook(connector: Connector, container: Container) -> Container
    build_customization_module = get_build_customization_module(connector)
    if hasattr(build_customization_module, "post_connector_install"):
            self.logger.info("Adding the post_connector_install hook to the base")
            container = await build_customization_module.post_connector_install(container)

    return container
    ```
    
    It would also give us a focused spot to document the connector lifecycle, what hooks are available? what do they do? how do you use them